packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -670,13 +675,12 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		this.checkNotDisposed(
 			"The parent branch has already been disposed and can no longer create new branches.",
 		);
+		this.editLock.checkUnlocked("Branching");
```

Comment:
Why don't we allow branching during events?

Reply:
It's because our public `nodeChanged` and `treeChanged` events currently fire as part of our internal `beforeChange` event (as opposed to `afterChange`). And `beforeChange` happens before the branch has updated its head to point to the commit with the new edit. So, currently, if you branch during a `nodeChanged`/`treeChanged` then you will be one commit behind the parent branch after the edit completes. 

We should probably fix that to make it actually work, but I'm not 100% sure we want to allow branching during events even if it does work. So for now, it's banned.

Reply:
Update: This problem was fixed in #23276. However, I've still kept the errors for now. We can later decide if we want to allow branching in callbacks and if so, it'll be trivial to remove the errors.